python/lsst/summit/utils/plotting.py

```diff
@@ -0,0 +1,256 @@
+    figure.colorbar(im, cax=cax, label="Counts")
+
+    if showCompass:
+        color = 'r'
+        try:
+            wcs = inputData.getWcs()
+        except AttributeError:
+            logger.warning("Failed to set WCS from input data. Compass will not be plotted.")
+            wcs = None
+
+        if wcs:
+            anchorRa, anchorDec = wcs.pixelToSky(compassLocation, compassLocation)
+            east = wcs.skyToPixel(geom.SpherePoint(anchorRa + 30.0 * geom.arcseconds, anchorDec))
+            north = wcs.skyToPixel(geom.SpherePoint(anchorRa, anchorDec + 30. * geom.arcseconds))
+            arrowLength = min(imageData.shape) * 0.05
+
+            an, bn = np.polyfit([compassLocation, north[0]], [compassLocation, north[1]], 1)
+            ae, be = np.polyfit([compassLocation, east[0]], [compassLocation, east[1]], 1)
+            pslope_n = -1. / an
+            pslope_e = -1. / ae
+
+            if north[0] > compassLocation:
+                xn = compassLocation + (arrowLength / np.sqrt(1 + pslope_n**2))
+            else:
+                xn = compassLocation - (arrowLength / np.sqrt(1 + pslope_n**2))
+            yn = pslope_n * (xn - compassLocation) + compassLocation
+
+            if east[0] > compassLocation:
+                xe = compassLocation + (arrowLength / np.sqrt(1 + pslope_e**2))
+            else:
+                xe = compassLocation - (arrowLength / np.sqrt(1 + pslope_e**2))
+            ye = pslope_e * (xe - compassLocation) + compassLocation
+
+            ax.arrow(compassLocation, compassLocation,
+                     xn-compassLocation, yn-compassLocation,
+                     head_width=1., head_length=1., color=color)
+            ax.arrow(compassLocation, compassLocation,
+                     xe-compassLocation, ye-compassLocation,
+                     head_width=1., head_length=1., color=color)
+
+            ax.text(xn, yn, 'N', color=color)
+            ax.text(xe, ye, 'E', color=color)
+
+    # Plot centroids
+    if centroid:
+        ax.plot(*zip(*centroid),
+                marker='o',
+                markeredgecolor='r',
+                markerfacecolor='None',
+                linestyle='None',
+                label='tupleList')
+
+    if sourceCat:
+        ax.plot(list(zip(sourceCat.getX(), sourceCat.getY())),
+                marker='^',
+                markeredgecolor='c',
+                markerfacecolor='None',
+                linestyle='None',
+                label='SrcCat')
+
+    if footprints:
+        match footprints:
+            case FootprintSet():
+                fs = FootprintSet.getFootprints(footprints)
+            case Footprint():
+                fs = [footprints]
+            case list():
+                try:
+                    footprints[0].getCentroid()
+                    fs = footprints
+                except AttributeError:
+                    raise AttributeError("Cannot get centroids for the first"
```

Comment:
I modified the error type and the message.

Reply:
That's much better, thanks.

Looking at it now though, I think it would be a lot more clear (and safe) to unpack the `fs` to get the `xy` inside the `match` block, and then work with that, rather than trying to unify the type to `fs` and then working with that. For example (and I know this is quite an edge case) but imagine there's a list where the first element is a footprint and the next isn't. Your code will fail on the `xy = [_.getCentroid() for _ in fs]` line, whereas if each `case` block extracted the `xy` instead, then you'd naturally get the correct error in the block there, and you wouldn't have to say "Failed on the first element" either, you could be specific. Does that make sense?

Reply:
I moved the` xy ` inside the `match` block. I also added info to the error message that prints the index of the element with an incorrect type.

Reply:
Thanks, looks great.